### PR TITLE
Fix / update links

### PR DESCRIPTION
**Pull request by bcantoni**
Found and fixed a bunch of links after running a link checker on the whole site.

### CODE REVIEW

**Comment by bcantoni** [comment]
@d45 for your review please.

For the fixed links I stuck with whatever style the file already had (full URL versus .md file), but we could update all of these to your style in a future PR if you want.

The only non link changes are:
* docs/trex_examples.md - fix indent so all samples line up correctly
* docs/trex_typescript.md - fix indent so that TOC appears correctly

**Comment by d45** [comment]
@bcantoni  Thanks for running the link checker and making the fixes! Which link checker did you use? (wc3?)
I'd like to incorporate a check as part of our GitLab and GitHub doc releases. 
 I keep forgetting about the UX design guide (we inherited this from an intern and added it to the API docs).

**Comment by bcantoni** [comment]
@d45 good point - let me document here for my own recall as well:

1. Install python version of https://linkchecker.github.io/linkchecker/: `pip install linkchecker`
2. Run local GH Pages equivalent of docs: `bundle install` followed by `bundle exec jekyll serve`
3. Run linkchecker against it: `linkchecker --ignore-url googleapis --ignore-url localhost --check-extern http://127.0.0.1:4000/extensions-api/`

The ignore for "googleapis" is because those were causing false alarms; the links work from a browser but not from the tool. The ignore for "localhost" is to avoid interpreting our "how to run locally" instructions as a real link.

**Comment by d45** [comment]
@bcantoni  - awesome! I'm sure we could incorporate this check with our GitLab docs (Embedding API v3, Hyper API, etc.). 
Running it manually for GitHub Help is another option before each PR. Thanks!

File: docs/trex_typescript.md [hunk 1]
```diff
@@ -100,4 +100,4 @@ In the HTML code for the extensions, the JavaScript files are referenced. The fo
 
 - [TypeScript in Visual Studio Code](https://code.visualstudio.com/docs/languages/typescript?=target="_blank")
 
-- [TypeScript Sample Extensions (GitHub)](https://github.com/tableau/extensions-api/tree/master/Samples-TypeScript) and [Use the TypeScript samples]({{site.baseurl}}/docs/trex_examples.html#use-the-typescript-samples)
+- [TypeScript Sample Extensions (GitHub)](https://github.com/tableau/extensions-api/tree/main/Samples-Typescript) and [Use the TypeScript samples]({{site.baseurl}}/docs/trex_examples.html#use-the-typescript-samples)
```
Comment (elbandito) [inline docs/trex_typescript.md:103]: This last link `Use the TypeScript samples` isn't working for me.  I'm guessing maybe it's okay, and there's a preprocessor that runs and evaluates the `{{site.baseurl}}` clauses?

File: docs/Style_Guidelines/ux_fonts.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ We recommend that you use a **sans serif font family** in your extension. Here,
 &nbsp;
 
 #### Benton Sans
-Benton Sans Book is the default font for the Tableau UI elements. <i>Learn more about Tableau font colors at **[Color](6 - Color.md)**.</i>
+Benton Sans Book is the default font for the Tableau UI elements. <i>Learn more about Tableau font colors at **[Color](ux_fonts.md)**.</i>
```
Comment (elbandito) [inline docs/Style_Guidelines/ux_fonts.md:34]: @bcantoni should this point to `ux_color.md`

Comment (bcantoni) [reply docs/Style_Guidelines/ux_fonts.md:34]: Good catch! I must have gotten a little too excited here.

File: docs/trex_typescript.md (hunk 1 above)
Comment (bcantoni) [reply docs/trex_typescript.md:103]: Yep, the site.baseurl format there is part of the GitHub Pages / Jekyll system and will be replaced by the suitable hostname when published. However, that means they are not viewable in a PR or when looking directly at code in the repo. One way you can do it is to do a local docs build on this branch, and then you can see how the links are modified. We have a short summary of how to do this over in the [TSC project](https://tableau.github.io/server-client-python/docs/dev-guide#running-docs-locally).

A future conversation with Dave is whether to make these links consistent throughout the docs and if so, which convention to follow.

File: docs/Style_Guidelines/ux_fonts.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ Using some of these common fonts that are likely available on your user's machin
 
 
 ###### Font Size
-It is important to compensate for OS and device speciﬁc DPI and font rendering. Since these are alternative fonts that aren't directly used in Tableau, you may choose to use different font sizes and weights based on your extension's design needs.
+It is important to compensate for OS and device specific DPI and font rendering. Since these are alternative fonts that aren't directly used in Tableau, you may choose to use different font sizes and weights based on your extension's design needs.
```
Comment (seanmakesgames) [inline docs/Style_Guidelines/ux_fonts.md:59]: hahaha how did this `fi` get in there? ﬁﬁﬁ

**Review by seanmakesgames — APPROVED** [review]
looks good to me!

**Comment by d45** [comment]
@bcantoni  Looks good. Got ahead and merge with `dev` if you aren't waiting on other changes. I need to update `ux_fonts.md` to add a reference to using Workbook filtering (Brian Carver pointed this out). I'm adding a new topic for that.

File: docs/Style_Guidelines/ux_fonts.md (hunk 2 above)
Comment (bcantoni) [reply docs/Style_Guidelines/ux_fonts.md]: Updated with this change.

**Pull request merged**